### DISCUSSION TITLE
Pebble watchface cannot be sent to watch by xdrip on Android 14

### DISCUSSION THREAD

**Discussion by ArthurusDent** [NightscoutFoundation/xDrip#3453] [General]
Hey everybody,

just wanted to report what to do when xdrip can't send the Pebble watchface to the Pebble watch on Android 14. 

I did search for a solution in some Facebook groups but couldn't find anything. 

To install the Pebble app on Android 14, this guide can be used: https://help.rebble.io/apk-load-android/#1

### The problem
But after installing the Pebble watchface can't be loaded onto the watch by xdrip. The error message is: "Couldn't load info about this app. Unable to install."

According to the logfiles, this is what's happening. The Pebble app doesn't have the necessary rights to access the Download folder.

```shell
05-09 15:21:50.507 22983 22983 E Pbl     : [MyPebbleFragment] handleArguments: sideLoadingUri was null
05-09 15:21:50.561  3582 17838 E Pbl     : [PebbleBundleManager] Failed to get local file for /storage/emulated/0/Download/xDrip-plus-pebble-trend-auto-install.pbw
05-09 15:21:50.561  3582 17838 E Pbl     : java.io.FileNotFoundException: /storage/emulated/0/Download/xDrip-plus-pebble-trend-auto-install.pbw (Permission denied)
``` 

### The solution
To give the Pebble app the necessary rights, open settings and follow the screenshots below:
![Screenshot_20240510-101537](https://github.com/NightscoutFoundation/xDrip/assets/6848629/9782c9e7-9432-4ca0-b559-b3263eb7e9a8)
![Screenshot_20240510-101544](https://github.com/NightscoutFoundation/xDrip/assets/6848629/783a2d35-58b1-4744-b2d6-df9d219ff4c3)
![Screenshot_20240510-101548](https://github.com/NightscoutFoundation/xDrip/assets/6848629/2a16e853-43b2-46c9-a3d3-0668c788a6b8)
![Screenshot_20240510-101558](https://github.com/NightscoutFoundation/xDrip/assets/6848629/be755aba-495b-4650-b449-28dfe91fe9e9)
![Screenshot_20240510-101602](https://github.com/NightscoutFoundation/xDrip/assets/6848629/29f2e855-bc44-4a41-b2f5-89d415e9f976)

### The result
Now you should be able to set the Pebble watchface in xdrip.

**Comment by jstevensog** [comment]
Well, the bad news is that the Pebble App is no longer supported on Android
14.  Had the same problem.
My trusty Pebble Time Steel is no more.
I have moved to a Garmin watch instead, just to get similar battery life.
I am happy enough with it, but not the web API delay.  Working on a push
based on the Pebble watchface I created years ago;
No timeline yet.
Cheers

On Fri, 10 May 2024 at 18:52, Arthurus Dent ***@***.***>
wrote:

> Hey everybody,
>
> just wanted to report what to do when xdrip can't send the Pebble
> watchface to the Pebble watch on Android 14.
>
> I did search for a solution in some Facebook groups but couldn't find
> anything.
>
> To install the Pebble app on Android 14, this guide can be used:
> https://help.rebble.io/apk-load-android/#1
> The problem
>
> But after installing the Pebble watchface can't be loaded onto the watch
> by xdrip.
>
> According to the logfiles, this is what's happening. The Pebble app
> doesn't have the necessary rights to access the Download folder.
>
> 05-09 15:21:50.507 22983 22983 E Pbl     : [MyPebbleFragment] handleArguments: sideLoadingUri was null
> 05-09 15:21:50.561  3582 17838 E Pbl     : [PebbleBundleManager] Failed to get local file for /storage/emulated/0/Download/xDrip-plus-pebble-trend-auto-install.pbw
> 05-09 15:21:50.561  3582 17838 E Pbl     : java.io.FileNotFoundException: /storage/emulated/0/Download/xDrip-plus-pebble-trend-auto-install.pbw (Permission denied)
>
> The solution
>
> To give the Pebble app the necessary rights, open settings and follow the
> screenshots below:
> Screenshot_20240510-101537.png (view on web)
> <https://github.com/NightscoutFoundation/xDrip/assets/6848629/9782c9e7-9432-4ca0-b559-b3263eb7e9a8>
> Screenshot_20240510-101544.png (view on web)
> <https://github.com/NightscoutFoundation/xDrip/assets/6848629/783a2d35-58b1-4744-b2d6-df9d219ff4c3>
> Screenshot_20240510-101548.png (view on web)
> <https://github.com/NightscoutFoundation/xDrip/assets/6848629/2a16e853-43b2-46c9-a3d3-0668c788a6b8>
> Screenshot_20240510-101558.png (view on web)
> <https://github.com/NightscoutFoundation/xDrip/assets/6848629/be755aba-495b-4650-b449-28dfe91fe9e9>
> Screenshot_20240510-101602.png (view on web)
> <https://github.com/NightscoutFoundation/xDrip/assets/6848629/29f2e855-bc44-4a41-b2f5-89d415e9f976>
> The result
>
> Now you should be able to set the Pebble watchface in xdrip.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/discussions/3453>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCCZ43AD3JPGNGFX4PDZATZBSDE5AVCNFSM6AAAAABHQH6UWSVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGYZTSNJVHA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
John Stevens
"You are how you live, not what you have."

**Reply by ArthurusDent** [reply]
> Well, the bad news is that the Pebble App is no longer supported on Android 14. 

Sure, but there's a way to still install the Pebble app on Android 14. That's what the first link is for. It uses the ADB to install the Pebble app. 

> I am happy enough with it, but not the web API delay.

I've been investigating Garmin watches because my Pebble watches won't keep working indefinitely, either because of hardware or software issues.

Never heard people complaining about the delay. How big is it?

**Comment by jstevensog** [comment]
> Sure, but there's a way to still install the Pebble app on Android 14.
That's what the first link is for. It uses the ADB to install the Pebble
app.
Install, yes.  Work, no.

> Never heard people complaining about the delay. How big is it?
Not more than a minute or two.  It is a limitation on the Garmin watch
polling xDrip.  It can only happen so often.
Push is the better way.
Cheers

On Fri, 17 May 2024 at 16:58, Arthurus Dent ***@***.***>
wrote:

> Well, the bad news is that the Pebble App is no longer supported on
> Android 14.
>
> Sure, but there's a way to still install the Pebble app on Android 14.
> That's what the first link is for. It uses the ADB to install the Pebble
> app.
>
> I am happy enough with it, but not the web API delay.
>
> I've been investigating Garmin watches because my Pebble watches won't
> keep working indefinitely, either because of hardware or software issues.
>
> Never heard people complaining about the delay. How big is it?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/discussions/3453#discussioncomment-9466573>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCCZ4YYFSMDBOD2QQ5JRF3ZCWTA7AVCNFSM6AAAAABHQH6UWSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TINRWGU3TG>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
John Stevens
"You are how you live, not what you have."

**Reply by ArthurusDent** [reply]
> Install, yes.  Work, no.

What exactly is happening? 
Do you get an error message? 
What have you tried to get it to work?

> Not more than a minute or two.

I see and good to know. Ok, that's a bit annoying and the Android watches probably show the new glucose reading right away, correct?

**Comment by jstevensog** [comment]
Android 14 has removed all support for the Pebble App, and Google, who
released a 64bit version a while ago, had not released a version that
works.  When you try and install it (it used to be available in the App
Store) in any way, you generally get an error message stating that the app
is too old and unsupported and will likely not work.  And that is what
happens.  It installs, but cannot connect to Bluetooth to connect to the
watch.  Didn't try to debug it any further, there was no point.

I cannot comment on how responsive the Android watches are.  I would assume
if the G6 is connected to the watch directly, it would be instantaneous,
and almost instantaneous if the phone is in between them.
I have never used an Android watch because the battery life it way too
low.  Even 48 hours is not sufficient for how I live.  At least, with the
Garmin, if I don't use the GPS or go silly tracking PulseOx etc, it will
last a full 7 days no issue.
Cheers

On Fri, 17 May 2024 at 20:50, Arthurus Dent ***@***.***>
wrote:

> Install, yes. Work, no.
>
> What exactly is happening?
> Do you get an error message?
> What have you tried to get it to work?
>
> Not more than a minute or two.
>
> I see and good to know. Ok, that's a bit annoying and the Android watches
> probably show the new glucose reading right away, correct?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/discussions/3453#discussioncomment-9469183>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCCZ42RTRGKJNXURTL74ITZCXOG3AVCNFSM6AAAAABHQH6UWSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TINRZGE4DG>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
John Stevens
"You are how you live, not what you have."

**Reply by ArthurusDent** [reply]
> When you try and install it (it used to be available in the App Store) in any way, you generally get an error message stating that the app is too old and unsupported and will likely not work.

The Pebble app is currently running on my Pixel 6a which is on Android 14 and it also did run on somebody's phone from my local looper group who uses a Pixel 7 with Android 14. Following the guide from my original post let him install the app. Bluetooth was no issue. If it is on your phone, there might be other software issues. 

The only problem we encountered was when trying to upload the xdrip watchface. That's why I have opened this discussion. There's a solution for this problem. 

Why would I go through the effort of writing about our experiences if the app wasn't installable and working in the first place?

> it will last a full 7 days no issue.

Yeah, that's one of the main reasons for me, too. And maybe the MIP display. The AMOLED watches look really nice but being able to always just read the watchface without having to rely on raise to awake is what's so nice about the Pebbles. The AMOLED versions can be put in an always on mode but that will hurt battery life. So, yeah, I'm undecided, but for the time being I still have two working Pebbles.

**Comment by jstevensog** [comment]
> The only problem we encountered was when trying to upload the xdrip
watchface. That's why I have opened this discussion. There's a solution for
this problem.

> Why would I go through the effort of writing about our experiences if the
app wasn't installable in the first place?

Apologies, then this is likely my Pixel 8 rather than the 6a or 7.  Worked
fine on my Pixel 5.  Tensor CPU rather than Qualcomm?  Who knows.

There should be no reason why sending the Pebble watchface app from xDrip+
does not work.  What are the log messages you are seeing when you try to do
this?

Yes, I wish other mainstream "smart" watches used ePaper displays.  I'd
likely get around a couple of weeks from a single watch charge if that were
the case, and have the "always on" display which I loved about the Pebble.
Sadly, "normal" people are too willing to put up with AMOLED so they can
see pretty pictures on a very small screen ;).  Not my cup of tea.

Cheers

On Mon, 20 May 2024 at 17:47, Arthurus Dent ***@***.***>
wrote:

> When you try and install it (it used to be available in the App Store) in
> any way, you generally get an error message stating that the app is too old
> and unsupported and will likely not work.
>
> The Pebble app is currently running on my Pixel 6a which is on Android 14
> and it also did run on somebody's phone from my local looper group who uses
> a Pixel 7 with Android 14. Following the guide from my original post let
> him install the app. Bluetooth was no issue. If it is on your phone, there
> might be other software issues.
>
> The only problem we encountered was when trying to upload the xdrip
> watchface. That's why I have opened this discussion. There's a solution for
> this problem.
>
> Why would I go through the effort of writing about our experiences if the
> app wasn't installable in the first place?
>
> it will last a full 7 days no issue.
>
> Yeah, that's one of the main reasons for me, too. And maybe the MIPS
> display. The AMOLED watches look really nice but being able to always just
> read the watchface without having to rely on raise to awake is what's so
> nice about the Pebbles. The AMOLED versions can be put in an always on mode
> but that will hurt battery life. So, yeah, I'm undecided, but for the time
> being I still have two working Pebbles.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/discussions/3453#discussioncomment-9493388>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCCZ4ZBQLQYENKY7O4ZNI3ZDGTBJAVCNFSM6AAAAABHQH6UWSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TIOJTGM4DQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
John Stevens
"You are how you live, not what you have."

**Reply by ArthurusDent** [reply]
> There should be no reason why sending the Pebble watchface app from xDrip+ does not work.  What are the log messages you are seeing when you try to do this?

Did you even read my initial post? I have solved this issue, that's why I opened this discussion. So that other people with the same problem could find this solution.

**Comment by williknow** [comment]
I could not get rid of the damn error message: "Couldn't load info about this app. Unable to install."
Installing the watchface from xdrip before visiting boot.rebble.io worked for me.

**Comment by tnharvey** [comment]
Wanted to share some help on this, in case anyone else comes across this. I completed the op's instructions but still received the following error when attempted to install the watchface from xDrip:

Error: file:///storage/emulated/0/Download/xDrip-plus-pebvle-trend-auto-install.pbw exposed beyond app through Intent.getData()

I dug a little and got nowhere, so I decided to try and install the file directly. This worked! Since xDrip already downloaded the file I needed, I installed the Rebble Sideloader app (my phone didn't have the option to open pbe files with Pebble). Google Play didn't let me install it, but I found the app in F-Droid and installed it that way.

Hope this helps!

**Reply by gh4tf** [reply]
@tnharvey thanks so much for this. It was the only thing that worked when I was trying to get the xDrip watchface back on my Pebble after switching to a new phone (Google Pixel 9A, Android 16). Thanks also to @ArthurusDent for the tips about allowing permissions for all files in the phone settings because I'm sure that helped too!
